impy/kinematics.py

```diff
@@ -81,22 +166,27 @@ def sort_list(k, idcs):
         self.component_Z = sort_list(self.component_Z, idcs)
         self.component_name = sort_list(self.component_name, idcs)
 
+    def get_maximum_AZ(self):
+        a_val = self.component_A
+        max_ind = a_val.index(max(a_val))
+        return self.component_A[max_ind], self.component_Z[max_ind]
+
     def get_random_AZ(self):
         """Return randomly an (A, Z) tuple according to the component fraction."""
         from numpy.random import choice
 
```

Comment:
Should these be public or private methods? I think they are private, no? By default, we make everything private.

Reply:
These methods are public because they are for use outside of `_FromParticleName` class

Reply:
Ok, but they should still be private, I think. In Python, "private" means a different thing to C++. The "private" parts of a library at not something the user should call, but they can still be "public"ly used within the library. So even if these methods are called by another impy code, they can still start with an underscore. The question is really whether it makes sense for users to call these methods. To me it looks like it is not something we want to expose to the users, because these methods are probably used to randomly sample from the CompositeTarget, so it is an implementation detail.

Reply:
I see. User should not use these methods. They are changed to private